PR for WebServer branch

AuroraLS3 opened the pull request:
To be merged after WebServer backend is functional

AuroraLS3 committed:
Create initial WebServer backend classes

AuroraLS3 committed:
Remove accidental license header additions

AuroraLS3 committed:
Added TODO to webserver branch required classes

cyberpwnn commented:
I would use a separate package for the web server stuff (your own perhaps) to keep as much out of pex itself and work as a layer on top of it if possible.

AuroraLS3 committed:
WebServer enabling. Requires config settings.

AuroraLS3 committed:
Moved new classes out of old Pex package space

AuroraLS3 committed:
Request and Response

AuroraLS3 committed:
Basic FileResponse

AuroraLS3 committed:
Created a React app (create-react-app) to see initial required responses

Node version 8.9.4
NPM version 5.6.0

Rest of the version information available in package.json

Build script will be created that copies the productions ready files to
src/main/resources/web-folder after running 'npm run build'

AuroraLS3 committed:
Started the Response system

AuroraLS3 committed:
Removed 1.7 configuration from maven-compiler-plugin

AuroraLS3 committed:
Begun work on RestAPI end point /api/users

AuroraLS3 committed:
JSONService Interfaces (Will need to be implemented) for all RestAPIs

AuroraLS3 committed:
Update JSONService javadocs

AuroraLS3 committed:
FileUtility javadoc

AuroraLS3 committed:
GET /api/users endpoint

cyberpwnn committed:
Merge branch 'legacy' into webserver

cyberpwnn committed:
Merge branch 'legacy' into webserver

Commit:
Merge branch 'legacy' into webserver

Commit:
Merge branch 'webserver' of https://github.com/VolmitSoftware/PermissionsEx into webserver

AuroraLS3 committed:
JSON formats for all JSON services that the frontend will use.

AuroraLS3 committed:
JSON parsing from Request body to RestAPIHandler

AuroraLS3 committed:
/api/users endpoints & Exceptions to JSONService interfaces.

AuroraLS3 committed:
Fixed /api/users/:uuid

AuroraLS3 committed:
/api/groups

AuroraLS3 committed:
/api/worlds

AuroraLS3 committed:
/api/plugins

AuroraLS3 committed:
Registered users, groups, worlds & plugins rest APIs to PageHandler

AuroraLS3 committed:
WorldJSON format inheritance to array instead of string.

AuroraLS3 committed:
Moved ParseException to proper package

AuroraLS3 committed:
Moved Request body parsing to Request.

AuroraLS3 committed:
Begun work on the login system

AuroraLS3 committed:
Config settings for WebServer

AuroraLS3 committed:
Reverted (auto-formatter tabs -> spaces) change from last commit

AuroraLS3 committed:
Temp fixed compile issue by introducing DummyJSONService

AuroraLS3 committed:
spaces -> tabs on the new files

AuroraLS3 committed:
Added a dependency for JSON Web Token (java-jwt)

AuroraLS3 committed:
JWT TokenVerifier

AuroraLS3 committed:
Token Authentication

AuroraLS3 committed:
Token generation to POST /api/login

Commit:
Merge branch 'legacy' into webserver

AuroraLS3 committed:
Added jBCrypt as a dependency (Password Salt+Hashing library)

AuroraLS3 committed:
Secret generation to TokenVerifier

AuroraLS3 committed:
BackupJSONService

AuroraLS3 committed:
Login Rest API and Register Rest API. Password hashing and registration.

AuroraLS3 committed:
Fixed some JSON formats (comments) and added dummy methods.

AuroraLS3 committed:
Moved most important class initialization to a single class

AuroraLS3 committed:
Replaced duplicate Wrapper class with AccessCallback

AuroraLS3 committed:
Fixed build failure by updating maven-shade-plugin to latest version:
Build failure was caused by a method reference
"[ERROR] Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:2.1:shade (default) on project PermissionsEx: Error creating shaded jar: Error in ASM processing class com/djrapitops/permissionsex/backends/web/http/Request.class: 52264 -> [Help 1]
"

AuroraLS3 committed:
FileConfig implementation for password has storage.

AuroraLS3 committed:
Relative css, js and json file serving

AuroraLS3 committed:
Hooked PexDashboard to PermissionsEx, Removed favicon

AuroraLS3 committed:
Fixed Frontend file serving from backend. Frontend installation info:
- Go to /react/dashboard/
- run 'npm install'
- run 'npm run build'
- copy all files from /react/dashboard/build/ to /src/main/resources/web/
- mvn install

AuroraLS3 committed:
Fixed webserver start displaying port twice

AuroraLS3 committed:
Webserver enable log

AuroraLS3 committed:
Fixed exceptions that were occurring in the web backend.

AuroraLS3 committed:
Removed token verification from login and register API

AuroraLS3 committed:
backend GET-method demo links to Frontend.

AuroraLS3 committed:
BackupJSONService should return json of created duplicate

AuroraLS3 committed:
User JSONService now uses names instead of UUIDs

AuroraLS3 committed:
Fixed improper tree traversal by TreePageHandler

AuroraLS3 committed:
Singular GET requests now parse spaces in names properly

AuroraLS3 committed:
Fixed typo in link

AuroraLS3 committed:
Backup API end point (/api/backups/)

AuroraLS3 committed:
Added built demo link app to resources/web
